The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6303287268/job/17112222269) [(plain)](https://github.com/rust-lang/rust/commit/b6d602eb3267a46226a8a0af82d3cb6b730f38a4/checks/17112222269/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_1bcb98e8-9679-4057-9caa-e0d0a2ed01ca
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=no-evil-sorting
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_1bcb98e8-9679-4057-9caa-e0d0a2ed01ca
GITHUB_REF=refs/pull/114454/merge
GITHUB_REF_NAME=114454/merge
GITHUB_REF_PROTECTED=false
---
89 
- error[E0408]: variable `b` is not bound in all patterns
-   --> $DIR/missing-bindings.rs:45:22
-    |
- LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
-    |                 -    ^^^^ pattern doesn't bind `b`
-    |                 variable not in all patterns
- 
98 error[E0408]: variable `a` is not bound in all patterns
99   --> $DIR/missing-bindings.rs:45:22
---
105 
+ error[E0408]: variable `b` is not bound in all patterns
+   --> $DIR/missing-bindings.rs:45:22
+    |
+ LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
+    |                 -    ^^^^ pattern doesn't bind `b`
+    |                 variable not in all patterns
+ 
+ 
106 error[E0408]: variable `e` is not bound in all patterns
108    |

111    |          |
112    |          pattern doesn't bind `e`
112    |          pattern doesn't bind `e`
113 
+ error[E0408]: variable `a` is not bound in all patterns
+   --> $DIR/missing-bindings.rs:45:33
+    |
+ LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
+    |              -                  ^^^^ pattern doesn't bind `a`
+    |              variable not in all patterns
+ 
114 error[E0408]: variable `b` is not bound in all patterns
115   --> $DIR/missing-bindings.rs:45:33
115   --> $DIR/missing-bindings.rs:45:33
116    |

136    |                            variable not in all patterns
137 
138 error[E0408]: variable `a` is not bound in all patterns
-   --> $DIR/missing-bindings.rs:45:33
-    |
- LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
-    |              -                  ^^^^ pattern doesn't bind `a`
-    |              variable not in all patterns
- 
- error[E0408]: variable `a` is not bound in all patterns
147   --> $DIR/missing-bindings.rs:61:29
147   --> $DIR/missing-bindings.rs:61:29
148    |
149 LL |                     Ok(a) | Err(_),
167 LL |                     B(b),
168    |                     ^^^^ pattern doesn't bind `a`
169 
- error[E0408]: variable `b` is not bound in all patterns
- error[E0408]: variable `b` is not bound in all patterns
+ error[E0408]: variable `a` is not bound in all patterns
171   --> $DIR/missing-bindings.rs:72:17
172    |
- LL |                     B(b),
-    |                       - variable not in all patterns
+ LL |                     A(_, a) |
175 ...
176 LL |                 B(_)
-    |                 ^^^^ pattern doesn't bind `b`
+    |                 ^^^^ pattern doesn't bind `a`
+    |                 ^^^^ pattern doesn't bind `a`
178 
- error[E0408]: variable `a` is not bound in all patterns
+ error[E0408]: variable `b` is not bound in all patterns
180   --> $DIR/missing-bindings.rs:72:17
181    |
- LL |                     A(_, a) |
+ LL |                     B(b),
+    |                       - variable not in all patterns
184 ...
185 LL |                 B(_)
---
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/or-patterns/missing-bindings.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/or-patterns/missing-bindings" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/or-patterns/missing-bindings/auxiliary" "--edition=2018"
stdout: none
--- stderr -------------------------------
error[E0408]: variable `beta` is not bound in all patterns
   |
   |
LL |     let (alpha | beta | charlie) = alpha; //~  ERROR variable `beta` is not bound in all patterns
   |          ^^^^^   ----   ^^^^^^^ pattern doesn't bind `beta`
   |          |       variable not in all patterns
   |          pattern doesn't bind `beta`


error[E0408]: variable `beta` is not bound in all patterns
   |
   |
LL |         Some(alpha | beta) => {} //~ ERROR variable `beta` is not bound in all patterns
   |              ^^^^^   ---- variable not in all patterns
   |              pattern doesn't bind `beta`

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:33:20
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:33:20
   |
LL |     let (A(a, _) | _) = X; //~ ERROR variable `a` is not bound in all patterns
   |            -       ^ pattern doesn't bind `a`
   |            variable not in all patterns

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:34:10
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:34:10
   |
LL |     let (_ | B(a)) = X; //~ ERROR variable `a` is not bound in all patterns
   |          ^     - variable not in all patterns
   |          pattern doesn't bind `a`

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:35:10
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:35:10
   |
LL |     let (A(..) | B(a)) = X; //~ ERROR variable `a` is not bound in all patterns
   |          |
   |          pattern doesn't bind `a`

error[E0408]: variable `a` is not bound in all patterns
error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:36:20
   |
LL |     let (A(a, _) | B(_)) = X; //~ ERROR variable `a` is not bound in all patterns
   |            -       ^^^^ pattern doesn't bind `a`
   |            variable not in all patterns

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:37:20
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:37:20
   |
LL |     let (A(_, a) | B(_)) = X; //~ ERROR variable `a` is not bound in all patterns
   |               -    ^^^^ pattern doesn't bind `a`
   |               variable not in all patterns

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:38:20
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:38:20
   |
LL |     let (A(a, b) | B(a)) = X; //~ ERROR variable `b` is not bound in all patterns
   |               -    ^^^^ pattern doesn't bind `b`
   |               variable not in all patterns

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:42:10
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:42:10
   |
LL |     let (A(A(..) | B(_), _) | B(a)) = Y; //~ ERROR variable `a` is not bound in all patterns
   |          |
   |          pattern doesn't bind `a`

error[E0408]: variable `a` is not bound in all patterns
error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:43:12
   |
LL |     let (A(A(..) | B(a), _) | B(A(a, _) | B(a))) = Y;
   |            |
   |            pattern doesn't bind `a`

error[E0408]: variable `c` is not bound in all patterns
error[E0408]: variable `c` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:12
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |            |
   |            pattern doesn't bind `c`

error[E0408]: variable `a` is not bound in all patterns
error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:22
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |              -       ^^^^ pattern doesn't bind `a`
   |              variable not in all patterns

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:22
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:22
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |                 -    ^^^^ pattern doesn't bind `b`
   |                 variable not in all patterns


error[E0408]: variable `e` is not bound in all patterns
   |
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |          |
   |          pattern doesn't bind `e`

error[E0408]: variable `a` is not bound in all patterns
error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:33
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |              -                  ^^^^ pattern doesn't bind `a`
   |              variable not in all patterns

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:33
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:33
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |                 -               ^^^^ pattern doesn't bind `b`
   |                 variable not in all patterns

error[E0408]: variable `c` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:33
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:33
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |                        -        ^^^^ pattern doesn't bind `c`
   |                        variable not in all patterns

error[E0408]: variable `d` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:33
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:33
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |                            -    ^^^^ pattern doesn't bind `d`
   |                            variable not in all patterns

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:61:29
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:61:29
   |
LL |                     Ok(a) | Err(_), //~ ERROR variable `a` is not bound in all patterns
   |                        -    ^^^^^^ pattern doesn't bind `a`
   |                        variable not in all patterns

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:68:21
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:68:21
   |
LL |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
   |                     ^^^^^^^ pattern doesn't bind `b`
LL |                     B(b), //~ ERROR variable `a` is not bound in all patterns

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:69:21
   |
   |
LL |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
   |                          - variable not in all patterns
LL |                     B(b), //~ ERROR variable `a` is not bound in all patterns
   |                     ^^^^ pattern doesn't bind `a`
error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:72:17
   |
   |
LL |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
...
Build completed unsuccessfully in 0:12:20
LL |                 B(_)
   |                 ^^^^ pattern doesn't bind `a`
   |                 ^^^^ pattern doesn't bind `a`

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:72:17
   |
LL |                     B(b), //~ ERROR variable `a` is not bound in all patterns
...
LL |                 B(_)
   |                 ^^^^ pattern doesn't bind `b`


error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:57:13
   |
LL | /             V1(
LL | |             //~^ ERROR variable `b` is not bound in all patterns
LL | |             //~| ERROR variable `c` is not bound in all patterns
...  |
...  |
LL | |                 B(Ok(a) | Err(a))
LL | |             ) |
   | |_____________^ pattern doesn't bind `b`
...
LL |                       B(b), //~ ERROR variable `a` is not bound in all patterns
...
LL |               V3(c),
   |               ^^^^^ pattern doesn't bind `b`


error[E0408]: variable `c` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:57:13
   |
LL | /             V1(
LL | |             //~^ ERROR variable `b` is not bound in all patterns
LL | |             //~| ERROR variable `c` is not bound in all patterns
...  |
...  |
LL | |                 B(Ok(a) | Err(a))
LL | |             ) |
   | |_____________^ pattern doesn't bind `c`
LL | /             V2(
LL | |                 A(
LL | |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
LL | |                     B(b), //~ ERROR variable `a` is not bound in all patterns
...  |
LL | |                 //~| ERROR variable `b` is not bound in all patterns
LL | |             ) |
   | |_____________^ pattern doesn't bind `c`
LL |               V3(c),

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:76:13
   |
   |
LL |                 B(Ok(a) | Err(a))
...
...
LL |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
...
LL |             V3(c),
   |             ^^^^^ pattern doesn't bind `a`

```

</details>

--- a/compiler/rustc_resolve/src/late.rs
+++ b/compiler/rustc_resolve/src/late.rs
@@ -41,9 +41,6 @@ type Res = def::Res<NodeId>;
 
 type IdentMap<T> = FxHashMap<Ident, T>;
 
-/// Map from the name in a pattern to its binding mode.
-type BindingMap = IdentMap<BindingInfo>;
-
 use diagnostics::{
     ElisionFnParameter, LifetimeElisionCandidate, MissingLifetime, MissingLifetimeKind,
 };
@@ -3164,8 +3161,8 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
     /// this is done hygienically. This could arise for a macro
     /// that expands into an or-pattern where one 'x' was from the
     /// user and one 'x' came from the macro.
-    fn binding_mode_map(&mut self, pat: &Pat) -> BindingMap {
-        let mut binding_map = FxHashMap::default();
+    fn binding_mode_map(&mut self, pat: &Pat) -> FxIndexMap<Ident, BindingInfo> {
+        let mut binding_map = FxIndexMap::default();
 
         pat.walk(&mut |pat| {
             match pat.kind {
@@ -3200,22 +3197,28 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
 
     /// Checks that all of the arms in an or-pattern have exactly the
     /// same set of bindings, with the same binding modes for each.
-    fn check_consistent_bindings(&mut self, pats: &[P<Pat>]) -> Vec<BindingMap> {
-        let mut missing_vars = FxHashMap::default();
-        let mut inconsistent_vars = FxHashMap::default();
+    fn check_consistent_bindings(
+        &mut self,
+        pats: &[P<Pat>],
+    ) -> Vec<FxIndexMap<Ident, BindingInfo>> {
+        // pats is consistent.
+        let mut missing_vars = FxIndexMap::default();
+        let mut inconsistent_vars = FxIndexMap::default();
 
         // 1) Compute the binding maps of all arms.
         let maps = pats.iter().map(|pat| self.binding_mode_map(pat)).collect::<Vec<_>>();
 
         // 2) Record any missing bindings or binding mode inconsistencies.
-        for (map_outer, pat_outer) in pats.iter().enumerate().map(|(idx, pat)| (&maps[idx], pat)) {
+        for (map_outer, pat_outer) in maps.iter().zip(pats.iter()) {
             // Check against all arms except for the same pattern which is always self-consistent.
-            let inners = pats
+            let inners = maps
                 .iter()
-                .enumerate()
+                .zip(pats.iter())
                 .filter(|(_, pat)| pat.id != pat_outer.id)
-                .flat_map(|(idx, _)| maps[idx].iter())
-                .map(|(key, binding)| (key.name, map_outer.get(&key), binding));
+                .flat_map(|(map, _)| map)
+                .map(|(key, binding)| (key.name, map_outer.get(key), binding));
+
+            let inners = inners.collect::<Vec<_>>();
 
             for (name, info, &binding_inner) in inners {
                 match info {
@@ -3244,10 +3247,7 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
         }
 
         // 3) Report all missing variables we found.
-        let mut missing_vars = missing_vars.into_iter().collect::<Vec<_>>();
-        missing_vars.sort_by_key(|&(sym, ref _err)| sym);
-
-        for (name, mut v) in missing_vars.into_iter() {
+        for (name, mut v) in missing_vars {
             if inconsistent_vars.contains_key(&name) {
                 v.could_be_path = false;
             }
@@ -3258,10 +3258,8 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
         }
 
         // 4) Report all inconsistencies in binding modes we found.
-        let mut inconsistent_vars = inconsistent_vars.iter().collect::<Vec<_>>();
-        inconsistent_vars.sort();
         for (name, v) in inconsistent_vars {
-            self.report_error(v.0, ResolutionError::VariableBoundWithDifferentMode(*name, v.1));
+            self.report_error(v.0, ResolutionError::VariableBoundWithDifferentMode(name, v.1));
         }
 
         // 5) Finally bubble up all the binding maps.

--- a/tests/ui/or-patterns/missing-bindings.stderr
+++ b/tests/ui/or-patterns/missing-bindings.stderr
@@ -103,22 +103,6 @@ LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
    |                 |
    |                 variable not in all patterns
 
-error[E0408]: variable `c` is not bound in all patterns
-  --> $DIR/missing-bindings.rs:45:33
-   |
-LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
-   |                        -        ^^^^ pattern doesn't bind `c`
-   |                        |
-   |                        variable not in all patterns
-
-error[E0408]: variable `d` is not bound in all patterns
-  --> $DIR/missing-bindings.rs:45:33
-   |
-LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
-   |                            -    ^^^^ pattern doesn't bind `d`
-   |                            |
-   |                            variable not in all patterns
-
 error[E0408]: variable `e` is not bound in all patterns
   --> $DIR/missing-bindings.rs:45:10
    |
@@ -143,6 +127,22 @@ LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
    |                 |
    |                 variable not in all patterns
 
+error[E0408]: variable `c` is not bound in all patterns
+  --> $DIR/missing-bindings.rs:45:33
+   |
+LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
+   |                        -        ^^^^ pattern doesn't bind `c`
+   |                        |
+   |                        variable not in all patterns
+
+error[E0408]: variable `d` is not bound in all patterns
+  --> $DIR/missing-bindings.rs:45:33
+   |
+LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
+   |                            -    ^^^^ pattern doesn't bind `d`
+   |                            |
+   |                            variable not in all patterns
+
 error[E0408]: variable `a` is not bound in all patterns
   --> $DIR/missing-bindings.rs:61:29
    |
@@ -151,14 +151,6 @@ LL |                     Ok(a) | Err(_),
    |                        |
    |                        variable not in all patterns
 
-error[E0408]: variable `a` is not bound in all patterns
-  --> $DIR/missing-bindings.rs:69:21
-   |
-LL |                     A(_, a) |
-   |                          - variable not in all patterns
-LL |                     B(b),
-   |                     ^^^^ pattern doesn't bind `a`
-
 error[E0408]: variable `b` is not bound in all patterns
   --> $DIR/missing-bindings.rs:68:21
    |
@@ -166,6 +158,14 @@ LL |                     A(_, a) |
    |                     ^^^^^^^ pattern doesn't bind `b`
 LL |                     B(b),
    |                       - variable not in all patterns
+
+error[E0408]: variable `a` is not bound in all patterns
+  --> $DIR/missing-bindings.rs:69:21
+   |
+LL |                     A(_, a) |
+   |                          - variable not in all patterns
+LL |                     B(b),
+   |                     ^^^^ pattern doesn't bind `a`
 
 error[E0408]: variable `a` is not bound in all patterns
   --> $DIR/missing-bindings.rs:72:17
@@ -184,6 +184,24 @@ LL |                     B(b),
 ...
 LL |                 B(_)
    |                 ^^^^ pattern doesn't bind `b`
+
+error[E0408]: variable `b` is not bound in all patterns
+  --> $DIR/missing-bindings.rs:57:13
+   |
+LL | /             V1(
+LL | |
+LL | |
+LL | |                 A(
+...  |
+LL | |                 B(Ok(a) | Err(a))
+LL | |             ) |
+   | |_____________^ pattern doesn't bind `b`
+...
+LL |                       B(b),
+   |                         - variable not in all patterns
+...
+LL |               V3(c),
+   |               ^^^^^ pattern doesn't bind `b`
 
 error[E0408]: variable `c` is not bound in all patterns
   --> $DIR/missing-bindings.rs:57:13
@@ -218,24 +236,6 @@ LL |                     A(_, a) |
 ...
 LL |             V3(c),
    |             ^^^^^ pattern doesn't bind `a`
-
-error[E0408]: variable `b` is not bound in all patterns
-  --> $DIR/missing-bindings.rs:57:13
-   |
-LL | /             V1(
-LL | |
-LL | |
-LL | |                 A(
-...  |
-LL | |                 B(Ok(a) | Err(a))
-LL | |             ) |
-   | |_____________^ pattern doesn't bind `b`
-...
-LL |                       B(b),
-   |                         - variable not in all patterns
-...
-LL |               V3(c),
-   |               ^^^^^ pattern doesn't bind `b`
 
 error: aborting due to 26 previous errors
 

--- a/tests/ui/resolve/resolve-inconsistent-names.stderr
+++ b/tests/ui/resolve/resolve-inconsistent-names.stderr
@@ -1,11 +1,3 @@
-error[E0408]: variable `a` is not bound in all patterns
-  --> $DIR/resolve-inconsistent-names.rs:13:12
-   |
-LL |        a | b => {}
-   |        -   ^ pattern doesn't bind `a`
-   |        |
-   |        variable not in all patterns
-
 error[E0408]: variable `b` is not bound in all patterns
   --> $DIR/resolve-inconsistent-names.rs:13:8
    |
@@ -13,6 +5,14 @@ LL |        a | b => {}
    |        ^   - variable not in all patterns
    |        |
    |        pattern doesn't bind `b`
+
+error[E0408]: variable `a` is not bound in all patterns
+  --> $DIR/resolve-inconsistent-names.rs:13:12
+   |
+LL |        a | b => {}
+   |        -   ^ pattern doesn't bind `a`
+   |        |
+   |        variable not in all patterns
 
 error[E0408]: variable `c` is not bound in all patterns
   --> $DIR/resolve-inconsistent-names.rs:19:9
@@ -54,6 +54,19 @@ LL |         (A, B) | (ref B, c) | (c, A) => ()
    |             |
    |             first binding
 
+error[E0408]: variable `Const2` is not bound in all patterns
+  --> $DIR/resolve-inconsistent-names.rs:31:9
+   |
+LL |         (CONST1, _) | (_, Const2) => ()
+   |         ^^^^^^^^^^^       ------ variable not in all patterns
+   |         |
+   |         pattern doesn't bind `Const2`
+   |
+help: if you meant to match on constant `m::Const2`, use the full path in the pattern
+   |
+LL |         (CONST1, _) | (_, m::Const2) => ()
+   |                           ~~~~~~~~~
+
 error[E0408]: variable `CONST1` is not bound in all patterns
   --> $DIR/resolve-inconsistent-names.rs:31:23
    |
@@ -67,19 +80,6 @@ note: you might have meant to match on constant `m::CONST1`, which exists but is
    |
 LL |     const CONST1: usize = 10;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^ not accessible
-
-error[E0408]: variable `Const2` is not bound in all patterns
-  --> $DIR/resolve-inconsistent-names.rs:31:9
-   |
-LL |         (CONST1, _) | (_, Const2) => ()
-   |         ^^^^^^^^^^^       ------ variable not in all patterns
-   |         |
-   |         pattern doesn't bind `Const2`
-   |
-help: if you meant to match on constant `m::Const2`, use the full path in the pattern
-   |
-LL |         (CONST1, _) | (_, m::Const2) => ()
-   |                           ~~~~~~~~~
 
 error[E0308]: mismatched types
   --> $DIR/resolve-inconsistent-names.rs:19:19

--- a/tests/ui/span/issue-39698.stderr
+++ b/tests/ui/span/issue-39698.stderr
@@ -1,3 +1,13 @@
+error[E0408]: variable `b` is not bound in all patterns
+  --> $DIR/issue-39698.rs:10:9
+   |
+LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
+   |         ^^^^^^^^^^^            -    ^^^^^^^^   ^^^^^^^^ pattern doesn't bind `b`
+   |         |                      |    |
+   |         |                      |    pattern doesn't bind `b`
+   |         |                      variable not in all patterns
+   |         pattern doesn't bind `b`
+
 error[E0408]: variable `c` is not bound in all patterns
   --> $DIR/issue-39698.rs:10:9
    |
@@ -7,16 +17,6 @@ LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}
    |         |             |                   variable not in all patterns
    |         |             pattern doesn't bind `c`
    |         pattern doesn't bind `c`
-
-error[E0408]: variable `d` is not bound in all patterns
-  --> $DIR/issue-39698.rs:10:37
-   |
-LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
-   |                  -          -       ^^^^^^^^   ^^^^^^^^ pattern doesn't bind `d`
-   |                  |          |       |
-   |                  |          |       pattern doesn't bind `d`
-   |                  |          variable not in all patterns
-   |                  variable not in all patterns
 
 error[E0408]: variable `a` is not bound in all patterns
   --> $DIR/issue-39698.rs:10:23
@@ -28,15 +28,15 @@ LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}
    |               |       pattern doesn't bind `a`
    |               variable not in all patterns
 
-error[E0408]: variable `b` is not bound in all patterns
-  --> $DIR/issue-39698.rs:10:9
+error[E0408]: variable `d` is not bound in all patterns
+  --> $DIR/issue-39698.rs:10:37
    |
 LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
-   |         ^^^^^^^^^^^            -    ^^^^^^^^   ^^^^^^^^ pattern doesn't bind `b`
-   |         |                      |    |
-   |         |                      |    pattern doesn't bind `b`
-   |         |                      variable not in all patterns
-   |         pattern doesn't bind `b`
+   |                  -          -       ^^^^^^^^   ^^^^^^^^ pattern doesn't bind `d`
+   |                  |          |       |
+   |                  |          |       pattern doesn't bind `d`
+   |                  |          variable not in all patterns
+   |                  variable not in all patterns
 
 error: aborting due to 4 previous errors
 

